sematic/resolvers/state_machine_resolver.py

```diff
@@ -40,14 +40,32 @@ def resolve(self, future: AbstractFuture) -> typing.Any:
 
             self._resolution_will_start()
 
+            graph_changed = False
```

Comment:
This `graph_changed` stuff is kind of an unrelated optimization for parallelism. If there are parallel branches with deeply nested funcs, work can get delayed un-necessarily because we hit the `_wait_for_scheduled_run` when there's more stuff that could be scheduled based on the completion of other fast DAG state changes.